opentimelineio_contrib/adapters/extern_rv.py

```diff
@@ -207,8 +211,23 @@ def _write_timeline(tl, to_session):
 def _create_media_reference(mr, to_session):
     if hasattr(mr, "media_reference") and mr.media_reference:
         if isinstance(mr.media_reference, otio.schema.ExternalReference):
-            to_session.setMedia([str(mr.media_reference.target_url)])
+
+            if mr.parent() and mr.parent().kind == otio.schema.TrackKind.Audio:
+                # Create blank video media to accompany audio for valid source
+                blank = "{},start={},end={},fps={}.movieproc".format(
+                    "blank",
+                    mr.available_range().start_time.value,
+                    mr.available_range().end_time_inclusive().value,
+                    mr.available_range().duration.rate
+                )
+                media = [blank, str(mr.media_reference.target_url)]
```

Comment:
Sure! I tested switching places on a previously created session file and it work nicely. I'll update now.

Reply:
Great! Did you notice how switching the order preserves the name of the audio source in the Session Manager? It hides the movieproc (unless you drill into the source), but promotes the relevant naming information.

There is supposed to be a way to achieve that in movieprocs to begin with, but one of my last significant additions to RV before my departure created a bug in this movieproc feature. I have started a conversation with Shotgun/AutoDesk to see if I can get them to fix my bug, but reordering here wasn't so bad.

Reply:
Yeah, that read much nicer. I thought for some reason that the order was important for the composition to work properly. Thanks for the tip.

Reply:
Hi @rogergodspeed and @apetrynet, bringing up this old pull request due to a weird edge case we came across here related to this. In a few instances, folks are setting a movie file in the "Audio" track, however this movie brings along some old reference video content. Depending on the stack setup, this reference video can show through. 

I am trying a few approaches to deal with this, but one simple solution I found was to swap the order here back to `media = [blank, str(mr.media_reference.target_url)]`. This seems "mute" the video (ie show only black) when the media_reference is pointing to a movie. We have had a few conversations on the intent of the "Audio" track kind, but one could argue that it should ONLY be for audio content, and ignoring (via "blanking" out) any video content is appropriate. 

This does bring back the question of the name in the session manager. We are getting around that by setting the clip item's name to something that makes sense for us, but I wonder if this code:

https://github.com/PixarAnimationStudios/OpenTimelineIO/blob/master/opentimelineio_contrib/adapters/extern_rv.py#L270

Would cause the ui.name to always be set (and not letting the "blank" movieproc though). 

In anycase, I put in #426 as a possible idea for switching the order again. Feel free to try and talk me out of this, or if you have alternate ideas, I would love to hear them.

Thanks,
Robyn

Reply:
Hi @avrata !

I unfortunately don't have enough knowledge of the inside workings of RV, so I can't really come up with other reasons why this would be a problem other than readability. 

That said, I think this edge case points at some other issues with OTIO that perhaps should be discussed. If I'm not mistaken, when looking at the media_reference for an audio item based on a movie file it points to a movie file and not an audio file or an audio track within that movie file. This might indicate that we should see this edge case quite often. One might solve this in a media linker, but when exporting from an NLE you often get the movie source as a media reference and most likely will run into this issue. 
There has been some discussion (#343) regarding a way to group media. Perhaps this is a good case for looking into this again?

Reply:
You are correct. Currently only the track kind specifies the type of media to expect. If a clip's media reference points to something of the wrong type, or a movie with both in it, the _intent_ is that only the media that matches the parent track's kind should be relevant. It would be reasonable for an adapter to mute any audio found in a video track, and to omit any video found in an audio track. However, since RV is happy to play both that doesn't really work well in practice. OTIO is missing a way for a media reference to select just one channel/stream within a media file - if we had that (and maybe an track kind of "Video+Audio"), then we could make this distinction clear.

Reply:
Every RVFileSource has a `group.noMovieAudio` property (http://www.tweaksoftware.com/static/documentation/rv/current/html/rv_reference.html#RVFileSource), but nothing analogous for video. The main concern I have with "muting" video this way is that it may not be respected when the compositing mode is changed. Anyway I am increasingly less and less familiar with how compositing is handled now that I have been gone over a year. I am taking steps to loop in someone that is still involved.

Reply:
Thanks. I did reach out to Kessler about this and he said they could turn the "onlyMovieAudio" (or whatever the inverse of the noMovieAudio flag is) into a feature request. This "blank'ing" of the video seemed like what would work now, but he did offer a warning to maybe do it twice to handle stereo modes.

Reply:
Hey all,

@avrata did reach out to me and @rogergodspeed just put me in touch with @jminor who pointed me here; sorry I didn't see this sooner!  With some discussion about the original problem with Jon, it sounds like the crux of the issue comes down to how the filename is being populated.

You should be able to use one (or two for stereo) blank movieprocs with the audio movie as the last item.  When you encode the filename, however, you will want to base64-encode the filename; once you do that it should be able to carry with it any name you wish.  If the filename looks like a path, the ui in the session manager will just show the extension-less basename (just as it would if it were a normal file path).

With base64 encoding, you can include all manner of characters that will pass fluidly through the filename attribute on the movieproc:

>>> import base64;base64.b64encode("Tr0ub4dor&3")
'VHIwdWI0ZG9yJjM='

```
/Applications/RV64.app/Contents/MacOS/RV64 blank,filename=VHIwdWI0ZG9yJjM=.movieproc
```

<img width="875" alt="screen shot 2019-02-20 at 5 49 33 pm" src="https://user-images.githubusercontent.com/1572121/53137517-e4d2bb80-3537-11e9-8afb-78eccbb12b4c.png">


Or with paths...

>>> import base64;base64.b64encode("/correct/horse/battery/staple.mov")
'L2NvcnJlY3QvaG9yc2UvYmF0dGVyeS9zdGFwbGUubW92'

```
/Applications/RV64.app/Contents/MacOS/RV64 blank,filename=L2NvcnJlY3QvaG9yc2UvYmF0dGVyeS9zdGFwbGUubW92.movieproc
```

<img width="894" alt="screen shot 2019-02-20 at 5 53 11 pm" src="https://user-images.githubusercontent.com/1572121/53137663-67f41180-3538-11e9-8af0-b7fa7387b8aa.png">


http://www.tweaksoftware.com/static/documentation/rv/current/html/rv_manual.html#File_Formats_Movie_File_Formats_RV_s_movieproc_Format___movieproc_


Between this and the using the blank movieproc first in the sources, I think you should be able to get both an audio-only placeholder along with a deterministic display name.